docker-src/cms/Dockerfile

```diff
@@ -1,57 +1,214 @@
 
 # Set timezone.
+ENV PHP_TIMEZONE America/New_York
 RUN echo "date.timezone = \"$PHP_TIMEZONE\"" > /usr/local/etc/php/conf.d/timezone.ini
 
-RUN a2enmod rewrite && a2enmod ssl && a2ensite default-ssl
+RUN cat /dev/urandom | LC_CTYPE=C tr -dc 'a-zA-Z0-9' | fold -w 64 | head -n 1 > /var/www/salt.txt
+
+WORKDIR /var/www
+
+RUN apk add --no-cache --virtual .build-deps \
+             # unknown needed
+             autoconf \
+             g++ \
+             pcre-dev \
+             libtool \
+             make \
+             curl \
+             git \
+             # Needed base depend
+             coreutils
+
+# Install PHP memcached extension
+# look at following for PHP 7.2 https://stackoverflow.com/a/41575677
+RUN set -xe \
+    && apk add --no-cache --update libmemcached-libs zlib mysql-client \
+    && apk add --no-cache --update --virtual .memcached-build-deps \
+               zlib-dev \
+               libmemcached-dev \
+               cyrus-sasl-dev \
+    && git clone -b php7 https://github.com/php-memcached-dev/php-memcached /usr/src/php/ext/memcached \
+    && docker-php-ext-configure /usr/src/php/ext/memcached \
+        --disable-memcached-sasl \
+    && docker-php-ext-install /usr/src/php/ext/memcached \
+    && rm -rf /usr/src/php/ext/memcached \
+    # Cleanup
+    && rm -rf /tmp/* ~/.pearrc /usr/share/php7 \
+    && docker-php-source delete \
+    && apk del .memcached-build-deps
+
+# install the PHP extensions we need
+# postgresql-dev is needed for https://bugs.alpinelinux.org/issues/3642
+RUN set -ex \
+  && apk add --no-cache --virtual .build-deps2 \
+             # GD depends
+             freetype-dev \
+             libjpeg-turbo-dev \
+             libpng-dev \
+             # mcrypt depends
+             libmcrypt-dev \
+             # xmlrpc depends
+             libxml2-dev \
+             libxslt-dev \
+             postgresql-dev \
+  # Configure and Install PHP extensions
+  && docker-php-ext-configure gd  \
+       --with-freetype-dir=/usr/include/ \
+       --with-jpeg-dir=/usr/include/ \
+       --with-png-dir=/usr/include/ \
+  && docker-php-ext-install -j "$(nproc)" \
+             gd \
+             iconv \
+             mcrypt \
+             mysqli \
+             opcache \
+             pdo_mysql \
+             pdo_pgsql \
+             xmlrpc \
+             xsl \
+             zip \
+  && runDeps="$( \
+    scanelf --needed --nobanner --format '%n#p' --recursive /usr/local \
+      | tr ',' '\n' \
+      | sort -u \
+      | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
+  )" \
+  && apk add --virtual .drupal-phpexts-rundeps $runDeps \
+  # Cleanup
+  && rm -rf /tmp/pear ~/.pearrc \
+  && chown -R www-data:www-data /usr/local/var/log \
+  && docker-php-source delete \
+  && apk del .build-deps .build-deps2 \
+  && rm -rf /tmp/* /var/cache/apk/*
+
+COPY docker-src/cms/php-conf.d/* /usr/local/etc/php/conf.d/
+
+COPY webroot /var/www/webroot
```

Comment:
This needs to be re-evaluated. When dockerfiles are poked to determine if a rebuild is necessary, as is the case with `make init` and `make update`, this line causes Docker to rebuild the images any time the environment is new. While this might seem like the preferred behavior, it adds a lot of time to the build process because docker doesn't use its cached images.

Additionally, once these files are written, this image becomes stale until another rebuild. If a user builds their environment on Monday, works on it through Friday without a `make init` or `make update`, when they do a `make export-prod` they will get Monday's code. Beyond this, the code we're copying in here from the `webroot` isn't enough to satisfy the installation as we are not copying in the `config` directory which would be necessary for this site to function on its own.

Reply:
Relating to your second part it contradicts your first part which I agree with. If a user built the images on a Monday. Made changes til Friday.  Then built prod. The cache would be invalidated due to the changes through the week.